base/compiler/abstractinterpretation.jl

```diff
@@ -887,6 +887,12 @@ function abstract_call_method_with_const_args(interp::AbstractInterpreter,
     if !const_prop_enabled(interp, sv, match)
         return nothing
     end
+    if is_total(result.effects)
```

Comment:
Now that I think about it, this is probably sufficient:
```suggestion
    if is_removable_if_unused(result.effects)
```